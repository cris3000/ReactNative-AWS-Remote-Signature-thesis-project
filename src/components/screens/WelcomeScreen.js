//React Imports
import React from 'react'

// React Native Imports
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image
} from 'react-native'

//Class For Welcome Screen, the first page a user will see once app is opened.
export default class WelcomeScreen extends React.Component {

  // Navigation to clicked link
  handleRoute = async (destination) => {
    await this.props.navigation.navigate(destination)
  }
  //Render method
  render() {
    return (

      // Container providing render dimensions of the app
      <View style={styles.container}>
    
        <TouchableOpacity 
        // On Press Button that will Send user to Login page
          onPress={() => this.handleRoute('SignIn')}
          style={styles.buttonStyle}>
          <Text style={styles.textStyle}>Login</Text>
        </TouchableOpacity>

        <TouchableOpacity 
         // On Press Button that will Send user to Register page
          onPress={() => this.handleRoute('SignUp')}
          style={styles.buttonStyle}>
          <Text style={styles.textStyle}>Register</Text>
        </TouchableOpacity>

        <TouchableOpacity 
         // On Press Button that will Send user to Forget Password page
          onPress={() => this.handleRoute('ForgetPassword')}
          style={styles.buttonStyle}>
          <Text style={styles.textStyle}>Forget password</Text>
        </TouchableOpacity>
        
      </View>
    )
  }
}
//Stylesheet 
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#b44666', // #b13366
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonStyle: {
    padding: 20,
  },
  textStyle: {
    fontWeight: 'bold',
    fontSize: 18,
    padding: 10,
    color: '#fff'
  }
})