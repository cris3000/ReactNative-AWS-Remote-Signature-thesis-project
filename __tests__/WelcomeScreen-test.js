// __tests__/HomeScreen-page-test.js
import 'react-native';
import React from 'react';
import WelcomeScreen from '../src/components/screens/WelcomeScreen';

import renderer from 'react-test-renderer';

// Testing whether screen renders
test('renders correctly', () => {
  const tree = renderer.create(
   <WelcomeScreen />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
