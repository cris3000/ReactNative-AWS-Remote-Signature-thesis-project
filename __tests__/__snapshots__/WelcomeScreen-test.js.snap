// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#b44666",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Sign In
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Forget password ?
    </Text>
  </View>
</View>
`;
